src/sqlparser/src/ast/mod.rs

```diff
@@ -2953,6 +2954,48 @@ impl fmt::Display for SqlOption {
     }
 }
 
+impl TryFrom<(&String, &String)> for SqlOption {
+    type Error = ParserError;
+
+    fn try_from((name, value): (&String, &String)) -> Result<Self, Self::Error> {
+        let mut tokenizer = Tokenizer::new(value);
```

Comment:
shall we reuse `parse_sql_option` for this case?

Reply:
The string must be constructed with 'with' to be usable, so parsing the map directly seems simpler.

Reply:
Here we handle CONNECTION specifically so maintaining the logic twice seems error prone to me.

Reply:
I found formatting wasn't hard, so I formatted first and then parse